proxy-robots-module/src/main/java/orwell/proxy/UdpBeaconFinder.java

```diff
@@ -0,0 +1,145 @@
+package orwell.proxy;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.*;
+import java.util.Enumeration;
+
+/**
+ * Created by Michaël Ludmann on 5/25/15.
+ */
+public class UdpBeaconFinder {
+    private final static Logger logback = LoggerFactory.getLogger(UdpBeaconFinder.class);
+    private final int broadcastPort;
+    private final int receiverBufferSize = 512;
+    private final DatagramSocket datagramSocket;
+    private final UdpBeaconDecoder udpBeaconDecoder;
+    private int maxAttemptsNumber = 1;
+    private int attemptsPerformed = 0;
+    private String pushAddress;
+    private String subscribeAddress;
+
+    public UdpBeaconFinder(final DatagramSocket datagramSocket, final int broadcastPort, final UdpBeaconDecoder udpBeaconDecoder) {
+        this.datagramSocket = datagramSocket;
+        this.broadcastPort = broadcastPort;
+        this.udpBeaconDecoder = udpBeaconDecoder;
+    }
+
+    public void setMaxAttemptsNumber(final int maxAttemptsNumber) {
+        if (1 > maxAttemptsNumber) {
+            logback.warn("Udp broadcasting attempts number cannot be less than 1");
+        } else {
+            this.maxAttemptsNumber = maxAttemptsNumber;
+        }
+    }
+
+    public int getNumberOfPerformedAttempts() {
+        return attemptsPerformed;
+    }
+
+    /**
+     * @return true if we did not get any meaningful response from server
+     * and have not reached max allowed attempts number
+     */
+    private boolean shouldTryToFindBeacon() {
+        if (null == udpBeaconDecoder) {
+            return attemptsPerformed < maxAttemptsNumber;
+        }
+        return (!udpBeaconDecoder.hasReceivedCorrectData() &&
+                attemptsPerformed < maxAttemptsNumber);
+    }
+
+    /**
+     * Find the server using UDP broadcast and fill data fields
+     */
+    public void startBroadcasting() {
+        try {
+            datagramSocket.setBroadcast(true);
+            while (shouldTryToFindBeacon()) {
+                logback.info("Trying to find UDP beacon, attempt [" + new Integer(attemptsPerformed+1) + "]");
+                // Broadcast the message over all the network interfaces
+                final Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+                while (interfaces.hasMoreElements()) {
+                    final NetworkInterface networkInterface = interfaces.nextElement();
+
+                    if (networkInterface.isLoopback() || !networkInterface.isUp()) {
+                        continue; // Do not broadcast to the loopback interface
+                    }
+
+                    for (final InterfaceAddress interfaceAddress : networkInterface.getInterfaceAddresses()) {
+                        final InetAddress broadcast = interfaceAddress.getBroadcast();
+                        if (null == broadcast) {
+                            continue;
+                        }
+
+                        // Send the broadcast package
```

Comment:
For instance, this command can nearly become the name of the method doing what comes next.